packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -528,9 +537,14 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
       return matrix.clone();
     }
 
+    final Offset alignedTranslation = widget.alignPanAxis && _panAxis != null
+      && _gestureType == _GestureType.pan
```

Comment:
Can this be done in `_onScaleUpdate` to avoid checking `_gestureType`?

Reply:
Good call 👍